### PR TITLE
Add GKE endpoints for sizes and disktypes

### LINKED ISSUE #10726: Add GKE endpoints for vmsizes and disktypes
### Description of the feature you would like to add / User story

As a user 
I would like to Add GKE endpoints for vmsizes and disktypes
in order to add that configuration for creating a default-nodepool while cluster creation

### Solution details
Endpoint (without credentials already exist), with credentials will be added.

- Requirement 1
- Requirement 2
- Requirement 3

### PR DESCRIPTION
**Pull request by imharshita**
Signed-off-by: Harshita sharma <harshita.sharma6174@gmail.com>

**What this PR does / why we need it**:
-  Add GKE endpoints for sizes and disktypes while cluster creation
-  Add more details to EKS subnets and securitygroups object

**Which issue(s) this PR fixes**:

Fixes #10726, #10750

**What type of PR is this?**

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Adds GET /api/v2/providers/gke/vmsizes and /api/v2/providers/gke/disktypes endpoint 
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

File: pkg/provider/cloud/gke/provider.go [hunk 1]
```diff
@@ -482,6 +482,59 @@ func createClient(ctx context.Context, serviceAccount string, scope string) (*ht
...
+		}
+		return nil
+	})
+
+	return sizes, err
+}
+
+func ListGKEDiskTypes(ctx context.Context, sa string, zone string) (apiv2.GKEDiskTypeList, error) {
+	diskTypes := apiv2.GKEDiskTypeList{}
+	// TODO: There are some issues at the moment with local-ssd and pd-extreme, that's why it is disabled at the moment.
```
Comment (xrstf) [inline pkg/provider/cloud/gke/provider.go]: What are those issues? Would be helpful for whoever wants to work on this TODO in 3, 6, 9 months time, so they can know if the issues are resolved already.

**Comment by imharshita** [comment]
@KhizerRehan you can refer to this PR, to reflect changes in dashboard code

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve ⚡

**Pull request merged**